public/app/core/components/AppChrome/MegaMenu/utils.ts

```diff
@@ -35,11 +35,17 @@ export const enrichWithInteractionTracking = (item: NavModelItem, megaMenuDocked
   const newItem = { ...item };
   const onClick = newItem.onClick;
   newItem.onClick = () => {
-    reportInteraction('grafana_navigation_item_clicked', {
-      path: newItem.url ?? newItem.id,
-      menuIsDocked: megaMenuDockedState,
-    });
-    onClick?.();
+    if (config.featureToggles.pinNavItems && newItem?.parentItem?.id === 'bookmarks') {
+      reportInteraction('grafana_navigation_pinned_item_clicked', {
+        path: newItem.url ?? newItem.id,
+      });
+    } else {
+      reportInteraction('grafana_navigation_item_clicked', {
```

Comment:
I don't know if this was discussed in any meeting, but could we make this a boolean in this interaction? Like the `menuIsDocked`, add a `itemIsBookmarked` or something? Then everything is still contained in just one interaction and we can filter it how we want.

Reply:
That's a good idea!